### PR TITLE
OrbitController: Add support for viewport transitions

### PR DESCRIPTION
**Pull request by 1chandu**
For #2046

#### Background
OrbitController: Add support for viewport transitions
Update `point-cloud-ply` to use linear transitions for rotation

#### Change List
- OrbitController: Add support for viewport transitions

### CODE REVIEW

File: modules/core/src/controllers/orbit-controller.js [hunk 1]
```diff
@@ -2,6 +2,15 @@ import {clamp} from 'math.gl';
 import Controller from './controller';
 import OrbitViewport from '../deprecated/orbit-viewport';
 import assert from '../utils/assert';
+import LinearInterpolator from '../transitions/linear-interpolator';
+import {TRANSITION_EVENTS} from './transition-manager';
+
+const LINEAR_TRANSITION_PROPS = {
+  transitionDuration: 300,
+  transitionEasing: t => t,
+  transitionInterpolator: new LinearInterpolator(),
```
Comment (Pessimistress) [inline modules/core/src/controllers/orbit-controller.js]: I think the default props include longitude/latitude?

Comment (1chandu) [reply modules/core/src/controllers/orbit-controller.js]: Right, all we need is to transition 'zoom' on double-tap, we can update this to `zoom` only and also update `_getTransitionProps` to return these props for double-tap only?

Comment (Pessimistress) [reply modules/core/src/controllers/orbit-controller.js]: What about keyboard navigation?

File: modules/core/src/controllers/orbit-controller.js [hunk 2]
```diff
@@ -293,4 +312,10 @@ export default class OrbitController extends Controller {
   constructor(props) {
     super(OrbitState, props);
   }
+
+  _getTransitionProps(event) {
+    // Enables Transitions on double-tap.
+    const props = event.type === 'doubletap' ? LINEAR_TRANSITION_PROPS : NO_TRANSITION_PROPS;
```
Comment (Pessimistress) [inline modules/core/src/controllers/orbit-controller.js]: Is this necessary? The transition would not be triggered unless zoom is changed.

**Pull request merged**